packages/core/contracts/financial-templates/perpetual-multiparty/ConfigStore.sol

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * @notice Returns current config or pending config if pending liveness has expired.
+     * @return ConfigSettings config settings that calling financial contract should view as "live".
+     */
+    function getCurrentConfig() external view override nonReentrantView() returns (ConfigSettings memory) {
+        if (_pendingProposalPassed()) {
+            return pendingConfig;
+        } else {
+            return currentConfig;
+        }
+    }
+
+    /**
+     * @notice Propose new configuration settings. New settings go into effect
+     * after a liveness period passes.
+     * @param newConfig Configuration settings to publish after `currentConfig.timelockLiveness` passes from now.
+     * @dev Callable only by owner. Calling this while there is already a pending proposal
+     * will overwrite the pending proposal.
+     */
+    function proposeNewConfig(ConfigSettings memory newConfig) external onlyOwner() nonReentrant() updateConfig() {
+        _validateConfig(newConfig);
+
+        // Warning: This overwrites a pending proposal!
+        pendingConfig = newConfig;
+
+        // Use current config's liveness period to timelock this proposal.
+        pendingPassedTimestamp = getCurrentTime() + currentConfig.timelockLiveness;
+
+        emit ProposedNewConfigSettings(
+            msg.sender,
+            newConfig.rewardRatePerSecond.rawValue,
+            newConfig.proposerBondPct.rawValue,
+            newConfig.timelockLiveness,
+            pendingPassedTimestamp
+        );
+    }
+
+    function publishPendingConfig() external nonReentrant() updateConfig() {}
+
+    /****************************************
+     *         INTERNAL FUNCTIONS           *
+     ****************************************/
+
+    // Check if pending proposal can overwrite the current config.
+    function _updateConfig() internal {
+        // If liveness has passed, publish new reward rate.
+        if (_pendingProposalPassed()) {
+            currentConfig = pendingConfig;
+
+            _deletePendingConfig();
+
+            emit ChangedNewConfigSettings(
+                currentConfig.rewardRatePerSecond.rawValue,
+                currentConfig.proposerBondPct.rawValue,
+                currentConfig.timelockLiveness
+            );
+        }
+    }
+
+    function _deletePendingConfig() internal {
+        delete pendingConfig;
+        pendingPassedTimestamp = 0;
+    }
+
+    function _pendingProposalPassed() internal view returns (bool) {
+        return (pendingPassedTimestamp != 0 && pendingPassedTimestamp <= getCurrentTime());
+    }
+
+    // Use this method to constrain values with which you can set ConfigSettings.
+    function _validateConfig(ConfigSettings memory config) internal pure {
+        // Make sure timelockLiveness is not too long, otherwise contract can might not be able to fix itself
+        // before a vulnerability drains its collateral. e.g. 604800 = 7 days.
```

Comment:
Oh! Is there a reason we don't want to set a lower bound on the timelock? Seems like it should be > a day or so...? Compound has [two days](https://etherscan.io/address/0x6d903f6003cca6255d85cca4d3b5e5146dc33925#code) as their absolute min, but I feel like 1 day is kinda a natural value...?

Thoughts?

Reply:
Yeah that's reasonable! done

Reply:
Did you forget to push?

Reply:
just pushed